src/Common/Commands.Common.Authentication/Authentication/AdalTokenProvider.cs

```diff
@@ -28,10 +28,10 @@ namespace Microsoft.Azure.Commands.Common.Authentication
     /// A token provider that uses ADAL to retrieve
     /// tokens from Azure Active Directory
     /// </summary>
-    public class AdalTokenProvider : ITokenProvider
+    public class AdalTokenProvider : IRenewableTokenProvider
     {
-        private readonly ITokenProvider userTokenProvider;
-        private readonly ITokenProvider servicePrincipalTokenProvider;
+        private readonly IRenewableTokenProvider userTokenProvider;
```

Comment:
I don't think we want to alter the ITokenProvider interface, or have these implement a new interface.  Rather, we should have them return a class the implements IRenewableToken

Reply:
Done